src/models.py

```diff
@@ -26,6 +27,7 @@ class ChunkType(str, Enum):
 class Chunk(BaseModel):
     """A unit part of a document."""
 
+    id: str
     text: str
```

Comment:
i noticed that the tests below are setting these IDs as incrementing ints, and the `parser_output_to_chunks` function is using the text block IDs... based on silly similar experiments i've done elsewhere, i have a feeling we might be about to hit some ID weirdness!

for example, what happens when a text block is converted to a chunk by `parser_output_to_chunks`, but the subsequent pipeline steps combine it with 3 of the surrounding text blocks, before splitting that combined thing into 8 new chunks? who gets the original text block ID?? are any of them really the original? what are we actually trying to identify with the `chunk.id`?

could the IDs be generated automatically based on the chunk's current content or history, rather than being manually set?

Reply:
i think they could, for sure! for now I was just delaying that decision until later, and needed chunks to have IDs as checking `chunk.heading == other_chunk` throws some sort of infinite recursion error 😆